### DISCUSSION TITLE
BdBrowser won't open at first launch

### DISCUSSION THREAD

**Discussion by SvenDixit** [tsukasa/BdBrowser#26] [Problems and issues]
BdBrowser used to work just fine, but after last night it suddenly didn't open when opening discord. I tried reinstalling, by downloading the most recent version and adding the unpacked folder to Edge, and expected that when opening discord, betterdiscord would open up, but nothing happened. 
Also in the settings, no additional settings for BD appeared. I tried manually downloading betterdiscord.asar and putting it in the scripts folder, but still no success. 
I found that on the "extensions" page, it shows the extension experiences some errors, which are the following;

Uncaught TypeError: Cannot read properties of undefined (reading 'Buffer')
Context: (different discord channels)
Stack Trace: js/frontend.js:2198 (anonymous function)

2198 const buffer = modules_discordmodules__WEBPACK_IMPORTED_MODULE_0__/* ["default"].Buffer.Buffer */ .Z.Buffer.Buffer;

I hope I gave enough info, and that this is something you can solve!

**Comment by tsukasa** [comment]
Greetings,

Thank you for your report! The source of this issue is Discord's removal of the Buffer module they previously webpacked. The non-existence of the module causes issues in BdBrowser's internals, thus preventing the proper initialization of the extension's injector. Your VFS data is unaffected by this.

BetterDiscord favours the idea of deprecating the use of the Buffer module with a grace period given by providing Node's implementation, going by the currently available GitHub commits.

Since the Buffer module is used throughout the extension, I will either have to find a substitute (i.e. from Browserify) or have to rework some parts.

I hope to have a fix ready sometime this weekend.
